wpilibj/src/athena/java/edu/wpi/first/wpilibj/SpeedControllerGroup.java

```diff
@@ -0,0 +1,111 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2016-2017. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Allows multiple {@link SpeedController} objects to be linked together.
+ */
+public class SpeedControllerGroup<T extends SpeedController> implements SpeedController {
+
+  private boolean m_isInverted = false;
+  private final List<T> m_speedControllers;
+
+  /**
+   * Create a new SpeedControllerGroup with no SpeedControllers.
+   */
+  public SpeedControllerGroup() {
+    m_speedControllers = new ArrayList<>();
+  }
+
+  /**
+   * Create a new SpeedControllerGroup with the provided SpeedControllers.
+   *
+   * @param speedControllers The SpeedControllers to add
+   */
+  public SpeedControllerGroup(Collection<T> speedControllers) {
```

Comment:
It needs to be generic because if it is not and I am using a CAN device and I want to call a method that is specific to that CAN device I would then need to cast the objects in the list to be CANTalons for example.